engine/src/execution_graph/logic_controllers/taskflow/executor.cpp

```diff
@@ -94,12 +115,44 @@ void task::run(cudaStream_t stream, executor * executor){
         }
         throw;
     }
+
+    auto task_result = kernel->process(std::move(input_gpu),output,stream, args);
+    
+    if(task_result.status == ral::execution::task_status::SUCCESS){
+        complete();
+    }else if(task_result.status == ral::execution::task_status::RETRY){
+        std::size_t i = 0;
+        for(auto & input : inputs){
+            if  (input->get_type() == ral::cache::CacheDataType::GPU || input->get_type() == ral::cache::CacheDataType::GPU_METADATA){
+                //this was a gpu cachedata so now its not valid
+                if(input_gpu.size() > 0 && i <= input_gpu.size()){
+                    static_cast<ral::cache::GPUCacheData *>(input.get())->set_data(std::move(input_gpu[i]));
```

Comment:
input_gpu is no longer valid here because we moved it.
Here we want to try to recover the data from task_result

Reply:
We also want to make sure that do add a null check to the inputs we are getting back from task_result, because we are not going to be sure that we did not actually do a std::move on them during the execution of the task. If it fails the null check, then we should do whatever we do when the task_status is a FAIL